### PR TITLE
Add Param "User Flight Profile"

### PR DESCRIPTION
**Pull request by thomasgubler**
This param has no effect within the Firmware.

The param can be used for example:

* in log post processing to have context about the flight
* in ground stations to adapt the UI automatically

**Test data / coverage**
I started sitl and checked that the param appears in QGC. I changed the param and uploaded the .ulg file to logs.px4.io and verified that the changed param appears in the analysis.

### CODE REVIEW

File: src/modules/commander/commander_params.c [hunk 1]
```diff
@@ -907,3 +907,20 @@ PARAM_DEFINE_INT32(COM_ASPD_FS_ACT, 0);
...
 PARAM_DEFINE_INT32(COM_ASPD_FS_DLY, 0);
 
+/**
+ * User Flight Profile
+ *
+ * Describes the intended use of the vehicle.
+ * Can be used by ground control software or log post processing.
+ * This param does not influence the behavior within the firmware. This means for example the control logic is independent of the setting of this param (but depends on other params).
+ *
+ * 0: Default
```
Comment (bkueng) [inline src/modules/commander/commander_params.c]:
```suggestion
 * @value 0 Default
```

File: src/modules/commander/commander_params.c [hunk 2]
```diff
@@ -907,3 +907,20 @@ PARAM_DEFINE_INT32(COM_ASPD_FS_ACT, 0);
...
+ * 100: Pro User
+ * 200: Flight Tester
+ * 300: Developer
+ *
+ * @min 0
```
Comment (bkueng) [inline src/modules/commander/commander_params.c]: You can drop the min/max

File: src/modules/commander/commander_params.c (hunk 1 above)
Comment (thomasgubler) [reply src/modules/commander/commander_params.c]: Thanks. Fixed in https://github.com/PX4/Firmware/pull/12187/commits/434aebd4fd394010a691d13b2174e6ae95928841

File: src/modules/commander/commander_params.c (hunk 2 above)
Comment (thomasgubler) [reply src/modules/commander/commander_params.c]: Thanks. Fixed in https://github.com/PX4/Firmware/pull/12187/commits/434aebd4fd394010a691d13b2174e6ae95928841

File: src/modules/commander/commander_params.c [hunk 3]
```diff
@@ -907,3 +907,18 @@ PARAM_DEFINE_INT32(COM_ASPD_FS_ACT, 0);
...
+ *
+ * Describes the intended use of the vehicle.
+ * Can be used by ground control software or log post processing.
+ * This param does not influence the behavior within the firmware. This means for example the control logic is independent of the setting of this param (but depends on other params).
+ *
+ * @value 0 Default
+ * @value 100 Pro User
+ * @value 200 Flight Tester
+ * @value 300 Developer
+ *
```
Comment (xdwgood) [inline src/modules/commander/commander_params.c]: Extra blank line :smiley:

Comment (thomasgubler) [reply src/modules/commander/commander_params.c]: I found that more readable and I see that for some of the other params in the file there is also a blank line and for others there is not. I am not opinionated. What is the standard here?

**Pull request merged**